src/google/protobuf/parse_context.h

```diff
@@ -929,15 +939,60 @@ template <typename T>
     return p + 2;
   }
   return VarintParseSlowArm(p, out, first8);
-#else   // __aarch64__
+#elif defined(__x86_64__) // __aarch64__
+  // Hot path: try to parse a 28 bit varint
+  // Range: 0..268,435,456
+  // Tagged: 0..33,554,432
+  //
+  // Messages longer than 128 bytes are not uncommon, so this
+  // should be faster in the general case
+
+  // Input is guaranteed atleast 10 bytes
+  uint32_t value = *reinterpret_cast<const uint32_t*>(p);
```

Comment:
We're seeing crashes here from alignment issues, does using memcpy affect the results?

Reply:
I couldn't find any good resources on the alignment requirements of C++ pointers. For x86_64, a memcpy should optimize to the same but I did notice `UnalignedLoad` having poor optimization on LLVM (though I could not remember if I chose a big endian target for this, but I opted out of everything but x86_64 since it's the only platform I've tested and which I have confidence in the results).

Are the crashes from the address sanitizer? Theres no reason why it should crash during normal runs.

Reply:
I'm seeing SIGILLs from clang, with and without sanitizers.  From what I found this is strictly UB in C++ due to the alignment requirements of uint32_t